api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -105,4 +134,21 @@ private boolean isAllergyInactive(CodeableConcept status) {
 		        .filter(c -> FhirConstants.ALLERGY_INTOLERANCE_CLINICAL_STATUS_VALUE_SET.equals(c.getSystem()))
 		        .anyMatch(c -> "inactive".equals(c.getCode()));
 	}
+	
+	private org.openmrs.User getCreator(Reference userReference) {
+		if (userReference == null) {
+			return null;
+		}
+		
+		if (userReference.getType() != null && !userReference.getType().equals("Practitioner")) {
+			throw new IllegalArgumentException("Reference must be to a Practitioner not a " + userReference.getType());
+		}
+		
+		String uuid = getReferenceId(userReference);
+		if (uuid == null) {
+			return null;
+		}
+		
+		return userDao.getUserByUuid(uuid);
+	}
 }
```

Comment:
Instead of doing this. Just inject practitionerRefereneceTranslator
```java
        @Inject
	private PractitionerReferenceTranslator<User> practitionerReferenceTranslator;
```
then 
 toFhirResource

`allergy.setRecorder(practitionerReferenceTranslator.toFhirResource(omrsAllergy.getCreator()));`

toOpenmrsType

`allergy.setCreator(practitionerReferenceTranslator.toOpenmrsType(fhirAllergy.getRecorder());`